onmt/bin/train.py

```diff
@@ -24,38 +26,85 @@
 
 def prepare_fields_transforms(opt):
     """Prepare or dump fields & transforms before training."""
-    DynamicArgumentParser.validate_prepare_opts(opt)
-
     transforms_cls = get_transforms_cls(opt._all_transform)
     specials = get_specials(opt, transforms_cls)
 
     fields = build_dynamic_fields(
         opt, src_specials=specials['src'], tgt_specials=specials['tgt'])
-    save_fields(opt, fields)
 
-    # prepare pretrained embeddings, if any
+    # maybe prepare pretrained embeddings, if any
     prepare_pretrained_embeddings(opt, fields)
 
-    transforms = make_transforms(opt, transforms_cls, fields)
-    save_transforms(opt, transforms)
+    if opt.dump_fields:
+        save_fields(fields, opt.save_data, overwrite=opt.overwrite)
+    if opt.dump_transforms or opt.n_sample != 0:
+        transforms = make_transforms(opt, transforms_cls, fields)
+    if opt.dump_transforms:
+        save_transforms(transforms, opt.save_data, overwrite=opt.overwrite)
     if opt.n_sample != 0:
         logger.warning(
             "`-n_sample` != 0: Training will not be started. "
             f"Stop after saving {opt.n_sample} samples/corpus.")
         save_transformed_sample(opt, transforms, n_sample=opt.n_sample)
-        sys.exit("Sample saved, please check it before restart training.")
+        logger.info(
+            "Sample saved, please check it before restart training.")
+        sys.exit()
+    return fields, transforms_cls
+
+
+def _init_train(opt):
+    """Common initilization stuff for all training replica."""
```

Comment:
Not sure "replica" is very explicit.
Maybe "process" / "subprocess" / "multiprocessing process" / "thread" ? (not sure which one)

Reply:
I'll take `process` I think.